Mathlib/Lean/SourceInfo_fromRef.lean

```diff
@@ -0,0 +1,19 @@
+/-
+Copyright (c) 2022 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+
+import Lean
+
+namespace Lean
+
+-- a variant of `SourceInfo.fromRef` in core
+
+/-- Constructs a synthetic SourceInfo using a ref : Syntax for the span. -/
```

Comment:
Could the doc-string explain that it differs from the one in core by having different (how?) behaviour when you set `synthetic := false`.

Reply:
I added a pointer to https://github.com/leanprover/lean4/pull/1620